### PR TITLE
feat: update merchant code validation

### PR DESCRIPTION
**Pull request by sebastianquek**
8 max limit alphanumeric (means if they enter 6 characters can go through too)

### CODE REVIEW

File: src/utils/validateMerchantCode.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export const merchantCodeRegex = /^[a-zA-Z0-9 _-]+$/; // Alphanumeric, space, underscore, dash
+export const merchantCodeRegex = /^[a-zA-Z0-9]{1,8}$/;
```
Comment (szenius) [inline src/utils/validateMerchantCode.tsx:1]: Why can't we use policy validationRegex for this?

Comment (sebastianquek) [reply src/utils/validateMerchantCode.tsx:1]: Yea was thinking we stick to this first, once the backend PR merges, we can change it!

File: src/utils/validateMerchantCode.test.tsx [hunk 2]
```diff
@@ -22,5 +28,8 @@ describe("validate merchant code", () => {
     expect(() => validateMerchantCode("å∫ç∂´ƒ©˙")).toThrow(
       "Invalid merchant code"
     );
+    expect(() => validateMerchantCode("ABCD12345678")).toThrow(
```
Comment (dalsontws) [inline src/utils/validateMerchantCode.test.tsx:31]: just my thoughts, maybe can cater for an edge case like 9 characters?
